meta-balena-common/recipes-support/balena-config-vars/balena-config-vars/os-config-json

```diff
@@ -0,0 +1,98 @@
+#!/bin/sh
+#
+# Copyright 2022 Balena Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+if [ -f "/usr/libexec/os-helpers-logging" ]; then
+	. /usr/libexec/os-helpers-logging
+else
+	error() {
+		echo "[ERROR] $*"
+	}
+	info() {
+		echo "[INFO] $*"
+	}
+fi
+
+CONF_TMPDIR=${CONF_TMPDIR:-"/var/volatile"}
+CONFIG_PATH=${CONFIG_PATH:-"/mnt/boot/config.json"}
+UNIT_CONF_PATH=${UNIT_CONF_PATH:-"/etc/systemd/unit-conf.json"}
+
+if [ ! -f "${UNIT_CONF_PATH}" ]; then
+	error "${UNIT_CONF_PATH} not found"
+	exit 1
+fi
+if [ ! -f "${CONFIG_PATH}" ]; then
+	error "${CONFIG_PATH} not found"
+	exit 1
+fi
+
+# Exit if syntactically incorrect
+jq -e '.' "${CONFIG_PATH}" > /dev/null
+jq -e '.' "${UNIT_CONF_PATH}" > /dev/null
+
+mkdir -p "${CONF_TMPDIR}"
+
+_tmpdir=$(mktemp -d)
+units=$(jq -r '.properties.units.properties | keys[]' "${UNIT_CONF_PATH}")
+for unit in ${units}; do
+	confs=$(jq -r ".properties.units.properties | .[\"${unit}\"].configuration" "${UNIT_CONF_PATH}")
+	for conf in ${confs}; do
+		_path=${conf%.*}
+		_value=${conf##*.}
```

Comment:
```suggestion
		_key=${conf##*.}
```
I found this variable name a little confusing, so maybe some comments would be helpful here?

```bash
# given
➜  conf="os.network.wifi.randomMacAddressScan"                                                               

➜  echo ${conf%.*}                                                                                               
os.network.wifi
➜  echo ${conf##*.}
randomMacAddressScan
```
So value is really the `key` we are interested in, or the last element the path we are watching, correct?

Reply:
hi @klutchell yes, it's the configuration `value` we are interested in, or the configuration `key` in JSON speech if you like.